File: www/cgi-bin/print.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void)
{
	char * operator;

	operator = getenv("OPERATOR");
	printf("HTTP/1.1 200 OK\r\n");
	printf("Content-Type: text/plain\r\n\r\n");
	if (operator[0] == 'a')
		printf("%.2f", atof(getenv("NUM1"))+atof(getenv("NUM2")));
	if (operator[0] == 's')
		printf("%.2f", atof(getenv("NUM1"))-atof(getenv("NUM2")));
	if (operator[0] == 'm')
		printf("%.2f", atof(getenv("NUM1"))*atof(getenv("NUM2")));
	if (operator[0] == 'd')
		printf("%.2f", atof(getenv("NUM1"))/atof(getenv("NUM2")));
	return 0;
}

File: www/cgi-bin/print.py
import os

operator = os.environ.get("OPERATOR")
num1 = float(os.environ.get("NUM1"))
num2 = float(os.environ.get("NUM2"))

if operator:
	if operator == "add":
		print("{:.2f}".format(num1 + num2))
	elif operator == "subtract":
		print("{:.2f}".format(num1 - num2))
	elif operator == "multiply":
		print("{:.2f}".format(num1 * num2))
	elif operator == "divide":
		if num2 != 0:
			print("{:.2f}".format(num1 / num2))
		else:
			print("inf")
	else:
		print("operator not found")
else:
	print("no operator set")

